### PR TITLE
Refactor monkey spec_unarmedattack

### LINKED ISSUE #71269: /datum/species/monkey/spec_unarmedattack() still exists and is a sin against god
https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/carbon/human/species_types/monkeys.dm#L77

image

we have code elsewhere to make monkeys attack using their heads now (which is something based on their brain, not their species), why is this still its own, separate proc?

image
also this line of code needs to be exorcised

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Refactors how the named proc works, the code was very soulful and I've fixed that

## Why It's Good For The Game

Squeaky clean code

Resolves https://github.com/tgstation/tgstation/issues/71269, kinda

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species_types/monkeys.dm [hunk 1]
```diff
@@ -75,38 +77,58 @@
...
 			return TRUE
-		for(var/datum/disease/bite_infection as anything in user.diseases)
-			if(bite_infection.spread_flags & (DISEASE_SPREAD_SPECIAL | DISEASE_SPREAD_NON_CONTAGIOUS))
-				continue
-			victim.ForceContractDisease(bite_infection)
+		return ..()
+
+	// this shouldn't even be possible, but I'm sure the check was here for a reason
+	if(!iscarbon(target))
 		return TRUE
```
Comment (Timberpoes) [inline code/modules/mob/living/carbon/human/species_types/monkeys.dm:90]: If it's possible but should never happen, you should stack_trace it so it fails unit tests (if such a unit test does or every may exist) and so coders can investigate and feex.

**Review by Timberpoes — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species_types/monkeys.dm (hunk 1 above)
Comment (ZephyrTFA) [reply code/modules/mob/living/carbon/human/species_types/monkeys.dm:90]:
```suggestion
	// this shouldn't even be possible, but I'm sure the check was here for a reason
	if(!iscarbon(target))
		stack_trace("HEY LISTEN! We are performing a species spec_unarmed attack with a non-carbon user. How did you fuck this up?")
		return TRUE
```

File: code/modules/mob/living/carbon/human/species_types/monkeys.dm [hunk 2]
```diff
@@ -75,38 +77,59 @@
...
+		var/mob/living/carbon/human/human_victim = victim
+		affecting = human_victim.get_bodypart(human_victim.get_random_valid_zone(even_weights = TRUE))
+	var/armor = victim.run_armor_check(affecting, MELEE)
+
+	if(prob(MONKEY_SPEC_ATTACK_BITE_MISS_CHANCE))
+		victim.visible_message( \
+			span_danger("[user]'s bite misses [victim]!"), \
+			span_danger("You avoid [user]'s bite!"), \
+			span_hear("You hear jaws snapping shut!"), \
+			COMBAT_MESSAGE_RANGE, user)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/monkeys.dm:107]: https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#multi-lining

File: code/modules/mob/living/carbon/human/species_types/monkeys.dm [hunk 3]
```diff
@@ -75,38 +77,59 @@
...
-	return ..()
+
+	var/obj/item/bodypart/head/mouth = user.get_bodypart(BODY_ZONE_HEAD)
+	if(!mouth) // check for them having a head, ala HARS
+		return TRUE
+
+	var/damage_roll = rand(mouth.unarmed_damage_low, mouth.unarmed_damage_high)
+	victim.apply_damage(damage_roll, BRUTE, affecting, armor)
+
+	victim.visible_message( \
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/monkeys.dm:118]: Same here

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**